src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -69,7 +69,12 @@ class IpexCommunicationService extends AgentService {
       .exchanges()
       .get(notifRecord.a.d as string);
     const credentialId = exn.exn.e.acdc.d;
-    await this.saveAcdcMetadataRecord(exn.exn.e.acdc.d, exn.exn.e.acdc.a.dt);
+    const connectionId = exn.exn.e.acdc.i;
```

Comment:
Does the `exn` message contain the sender somewhere? This `acdc.i` field is the issuer. They will most likely be the same, but not necessarily.

Reply:
I check the ConnectionId field is acdc.i, we are saving the connection record as sender

Reply:
I check the ConnectionId field is acdc.i, we are saving the connection record as sender

Reply:
`acdc.i` is the issuer, not the sender of the `/exn/grant` message. More often than not, these are the same.

However, there might be tunneling in the future where an issuer uses a different AID to send issue the credential (public) than the one used to send the grant message. This is for privacy reasons.

So wherever we relate the connectionId of to an ACDC, it should be the sender of the exn message, not the issuer (`acdc.i`).

Reply:
I see three field had the same value: `exn.e.acdc.i`, `exn.i`, `exn.e.anc.i`
Not sure What is connectionId?

Reply:
I guess that is `exn.e.anc.i`, because I think the the main field related to issuer, not connection

Reply:
@bao-sotatek `exn.i` would be correct - this is the sender of the exn message which is what we are looking for. `anc` is related to the signature on `acdc`